uf-mil/mil
# Darknet ros branch

To test:
1. Check out darknet_ros_branch and update your submodules.
2. Make sure to compile (if it fails for whatever reason, do a full compile by deleting your build and devel)
3. Run ```roslaunch navigator_launch vrx.launch run_task:=True world:=perception_task --screen```
4. You can see the bounding boxes by opening rqt and seeing the ```darknet_ros/detection_image```
5. You can also see the bounding box values from the ```/darknet_ros/bounding_boxes``` topic
6. Alternatively (cancel the simulation), run the rostest for darknet_ros which should pass: 
```catkin_make run_tests_darknet_ros```

I am thinking about modifying the image topic darknet_ros listens to such that it only processes an image when we want it to (as opposed to every image that comes from the left_camera). Let me know if it works and what you think!

Do not merge until we have decided we want to use darknet over tensorflow.
